src/homepageExperience/ClickThroughAnnouncementHandler.tsx

```diff
@@ -0,0 +1,67 @@
+// Libraries
+import {FC, useEffect} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
+import {useLocalStorageState} from 'use-local-storage-state'
+
+// Selectors
+import {selectCurrentIdentity} from 'src/identity/selectors'
+
+// Utils
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Constants
+import {CLOUD} from 'src/shared/constants'
+
+export const ClickThroughAnnouncementHandler: FC = () => {
+  const dispatch = useDispatch()
+  const {account} = useSelector(selectCurrentIdentity)
+
+  const [announcementState, setAnnouncementState] = useLocalStorageState(
+    'clickThroughAnnouncement',
+    {
+      announcementID: '',
+      display: true,
+    }
+  )
+
+  const setCurrentAnnouncement = (announcementID: string): void => {
+    if (announcementState['announcementID'] !== announcementID) {
+      setAnnouncementState({
+        announcementID: announcementID,
+        display: true,
+      })
+    }
+  }
+
+  const handleDismissAnnouncement = (): void => {
+    setAnnouncementState(prevState => ({
+      ...prevState,
+      display: false,
+    }))
+    dismissOverlay()
+  }
+
+  useEffect(() => {
+    // Current Announcement: PAYG Pricing Increase
+    // Audience: Cloud, Pay As You Go, Direct Signups
+    const isPaygAccount = account.type === 'pay_as_you_go'
+    const isDirectSignup = account.billingProvider === 'zuora'
+    const isTargetAudience = CLOUD && isPaygAccount && isDirectSignup
+
```

Comment:
I think it may be best to return out of the component immediately if not `CLOUD`, as this is a homepage/firstmile component that can run in OSS. In OSS, the account.type and account.billingProvider properties don't exist, so ideally we would return before checking those to avoid potential errors.

Reply:
moved & fetched